### DISCUSSION TITLE
Reimplementing `PointCloud.segment_plane` on Python

### DISCUSSION THREAD

**Discussion by evbernardes** [isl-org/Open3D#6389] [Q&A]
I am trying to re-implement `PointCloud.segment_plane` in pure Python following the C++ implementation [here](https://github.com/isl-org/Open3D/blob/master/cpp/open3d/geometry/PointCloudSegmentation.cpp). The reason for that, is that as a second step I am going to create other types of segmentation methods that follow a similar procedure. I imagine this is the only implementation available since the Python hooks are generated automatically, am I right?

However, I haven't been able to find a way to call many of its internal methods/classes on Python. For example, the Python equivalents of `RandomSampler`, or `EvaluateRANSACBasedOnDistance`.

Is there a way for me to call these functions, or are the hooks for internal functions like those not generated?

**Comment by saurabheights** [comment]
@evbernardes The hooks are not generated automatically. There is `/Open3d/cpp/pybind/` directory which generates python bindings for code in `/Open3d/cpp/open3d/`.

For e.g. - See https://github.com/isl-org/Open3D/blob/master/cpp/pybind/geometry/pointcloud.cpp#L199 binds C++ method `PointCloud::SegmentPlane` to new python method `segment_plane`.

I did quick search, but it looks like there is **no** binding for `RandomSampler` or `EvaluateRANSACBasedOnDistance`.

P.S. - Feel free to add a PR for this.
